### DISCUSSION TITLE
UI project

### DISCUSSION THREAD

**Discussion by Etrinque** [henomis/lingoose#189] [General]
Hello Lingoose devs, 
I am currently working on a career path trajectory change, I have spent the better part of the last year learning Software development and programming, Namely GO, it has been one of the most eye opening experiences I have had professionally and I cannot wait to begin working in the field.  That being said its time for me to develop a portfolio project of my own outside of the course work I have done, i.e. namely Boot.Dev.

A little overview - 
For my capstone project I have had the idea floating around in my head to build a moderately capable GUI for running local LLM/AI on homelabs/NAS/etc. The general scope of the project would entail a Godot front end with some simple UI elements for chosing a Model and a DB/Storage location to interact with said models and save session data. And a menu which will have user pref. for the gui and some other minor controls. Overall a simple platform to interact with to make working with local models a little more approachable.

The goal would be being able to have multiple models locally on your own server at your disposal on a whim and to be able to recall past sessions to continue where you left off in a simple and hopefully compact (as compact as I can achieve) implementation would be of use to myself, and maybe others.  Ideally I would like to build the bridge to make this all happen.  Which requires some relatively obscure interactions between GDScript/C++/C# (but thats for me to figure out) and the backend. Maybe via WebSockets for communication from host to homelab/server. I may be digressing here. 

With that, then comes the backend side, I was looking into options for GO development in the LLM/AI world and stumbled upon Lingoose, which appears to be the framework I would need to start building the application on top of.

Structure would look something like this:
Docker Container-> Godot Front End -> Lingoose/Custom -> Local-AI -> Models -> DB 
So the Lingoose/Custom implementation is the effective pivot point that will tie this all together. 

All this to say I am excited to work with your project and to begin implementing an integration with it, so i greatly appreciate the work you have done.  It may be too ambitious and I may be way in over my head and am sure I will have questions as they arise but for now I just wanted to spitball my idea out into the lingoose-verse. 

If anyone has any ideas or knows any limitations I may encounter from my short description please let me know.

**Comment by henomis** [comment]
It looks like a great project. Feel free to ask questions :rocket:
